src/ResourceManagement/Compute/EncryptionExtensionIdentifier.cs

```diff
@@ -0,0 +1,107 @@
+
+        /// <summary>
+        /// Return encryption extension publisher name.
+        /// </summary>
+        /// <returns>encryption extension publisher name</returns>
+        internal static string PublisherName()
+        {
+            return ENCRYPTION_EXTENSION_PUBLISHER;
+        }
+
+        /// <summary>
+        /// Return OS specific encryption extension type.
+        /// </summary>
+        /// <param name="osType">OS Type</param>
+        /// <returns>OS specific encryption extension type</returns>
+        internal static string TypeName(OperatingSystemTypes osType)
+        {
+            if (osType == OperatingSystemTypes.Linux)
+            {
+                return LINUX_ENCRYPTION_TYPE_NAME;
+            }
+            else
+            {
+                return WINDOWS_ENCRYPTION_TYPE_NAME;
+            }
+        }
+
+        /// <summary>
+        /// Given os type and no aad flag return the encryption extension version.
+        /// </summary>
+        /// <param name="osType">OS Type</param>
+        /// <param name="isNoAAD">no aad flag</param>
+        /// <returns>the encryption extension version</returns>
+        internal static string Version(OperatingSystemTypes osType, bool isNoAAD)
+        {
+            if (osType == OperatingSystemTypes.Linux)
+            {
+                return isNoAAD ? LINUX_NOAAD_ENCRYPTION_EXTENSION_VERSION : LINUX_LEGACY_ENCRYPTION_EXTENSION_VERSION;
+            }
+            else
+            {
+                return isNoAAD ? WINDOWS_NOAAD_ENCRYPTION_EXTENSION_VERSION : WINDOWS_LEGACY_ENCRYPTION_EXTENSION_VERSION;
+            }
+        }
+
+        /// <summary>
+        /// Checks whether the given version is a legacy extension version or no-aad extension version
+        /// for the given OS type.
+        /// </summary>
+        /// <param name="osType">OS Type</param>
+        /// <param name="version">extension version</param>
+        /// <returns>true if no-aad</returns>
+        internal static bool IsNoAADVersion(OperatingSystemTypes osType, string version)
+        {
+            string majorVersion = version.Split('.')[0];
+            if (osType == OperatingSystemTypes.Linux)
+            {
+                return majorVersion.Equals(LINUX_NOAAD_ENCRYPTION_EXTENSION_VERSION.Split('.')[0]);
+            }
+            else
+            {
+                return majorVersion.Equals(WINDOWS_NOAAD_ENCRYPTION_EXTENSION_VERSION.Split('.')[0]);
+            }
+        }
+
+        /// <summary>
+        /// Checks whether the given publisher name is encryption publisher name.
+        /// </summary>
+        /// <param name="publisherName">publisher name</param>
+        /// <returns>true if the given publisher name is encryption publisher name</returns>
+        internal static bool IsEncryptionPublisherName(string publisherName)
+        {
+            return publisherName.Equals(PublisherName(), System.StringComparison.OrdinalIgnoreCase);
```

Comment:
1. This is a static method, which assume the PublishName() is not good coding practice. 
2. Null check for input value.

Reply:
I don't know why we cannot use a static method in another. What should I do if I want to split two method?